libyul/Object.h

```diff
@@ -67,8 +67,23 @@ struct Object: ObjectNode
 	std::string toString(Dialect const* _dialect) const override;
 
 	/// @returns the set of names of data objects accessible from within the code of
-	/// this object.
-	std::set<YulString> dataNames() const;
+	/// this object, including the name of object itself
+	std::set<YulString> qualifiedDataNames() const;
+
+	/// @returns vector of subIDs if possible to reach subobject with @a _qualifiedName, throws otherwise
+	/// For "B.C" should return vector of two values if success (subId of B and subId of C in B).
+	/// In object "A" if called for "A.B" will return only one value (subId for B)
+	/// will return empty vector for @a _qualifiedName that equals to object name.
+	/// Example:
+	/// A1 { B2 { C3 D3 }, E2 { F3 { G4 K3 H4 { I5 } } } }
+	/// pathToSubObject("A1.E2.F3.H4") == {1, 0, 2}
+	/// pathToSubObject("E2.F3.H4") == {1, 0, 2}
+	/// pathToSubObject("A1") == {}
+	/// It doesn't accept data (will throw)
```

Comment:
I'd rephrase it a bit to make it easier to realize what "data" is in this context:
```suggestion
	/// The path must not lead to a @a Data object (will throw in that case).
```